# Curso de Introducción a la Ciencia de Datos

**Ejercicio SF Salaries**
Utilizaremos el `SF Salaries Dataset` Kaggle. Simplemente siga y complete las tareas que se detallan a continuación en negrita. Las tareas serán cada vez más difíciles a medida que avance.

La idea principal es aplicar pandas para organizar informacion, manipularla, buscar información en el dataset


**Importe pandas como pd. e imprima la version que esta usando**

In [ ]:
# Copie el código aca

In [ ]:
import pandas as pd
pd.__version__


'2.2.2'

*La versión de panda es 2.2.2*

**leer el archivo Salaries.csv en un dataframe llamado salarios**

se encuentra en esta url y se puede leer directamente con pandas

https://raw.githubusercontent.com/dpalacioj/DataAI-Fundamentos-Aplicaciones/main/Analitica/ciencia_datos_con_python/tareas/datos/Salaries.csv


In [ ]:
# Copie el código aca
# Carga del dataframe
url = "https://raw.githubusercontent.com/dpalacioj/DataAI-Fundamentos-Aplicaciones/main/Analitica/ciencia_datos_con_python/tareas/datos/Salaries.csv"
df = pd.read_csv(url)


**Use el método `.info()` para encontrar cuantas Columnas de atributos hay y que tipo de datos tiene cada una.**

In [ ]:
# Copie el código aca

*El dataframe contiene 13 columnas, la misma cantidad de filas 148654*

In [ ]:
#Información de cuantas columnas hay y tipo de datos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148654 entries, 0 to 148653
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Id                148654 non-null  int64  
 1   EmployeeName      148654 non-null  object 
 2   JobTitle          148654 non-null  object 
 3   BasePay           148045 non-null  float64
 4   OvertimePay       148650 non-null  float64
 5   OtherPay          148650 non-null  float64
 6   Benefits          112491 non-null  float64
 7   TotalPay          148654 non-null  float64
 8   TotalPayBenefits  148654 non-null  float64
 9   Year              148654 non-null  int64  
 10  Notes             0 non-null       float64
 11  Agency            148654 non-null  object 
 12  Status            0 non-null       float64
dtypes: float64(8), int64(2), object(3)
memory usage: 14.7+ MB


- Segun la información del dataframe su tamaño es de *memory usage: 14.7+ MB*

- las columnas **Notes** y **Status** solamente tienen valores *NaN*, verifique que dice **0 non-null** osea 0 valores son diferentes a *NaN*
entonces todos los valores son *NaN* en esas columnas

**Elimine las columnas *Notes* y *Status* Recuerde usar el método de forma permanente (inplace), y luego visualice cuantas filas y columnas tiene luego de eliminar las columnas**

In [ ]:
# Copie el código aca

In [ ]:
# Eliminación de columnas
df.drop(columns= ["Notes", "Status"], inplace=True)

# Dimensiones del DataFrame
df.shape

(148654, 11)

*Después de eliminar las 2 columnas, el dataframe continua con 11 columnas, la misma cantidad de filas 148654*

**¿Cual es la nueva información del dataframe?**

In [ ]:
# Copie el código aca

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148654 entries, 0 to 148653
Data columns (total 11 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Id                148654 non-null  int64  
 1   EmployeeName      148654 non-null  object 
 2   JobTitle          148654 non-null  object 
 3   BasePay           148045 non-null  float64
 4   OvertimePay       148650 non-null  float64
 5   OtherPay          148650 non-null  float64
 6   Benefits          112491 non-null  float64
 7   TotalPay          148654 non-null  float64
 8   TotalPayBenefits  148654 non-null  float64
 9   Year              148654 non-null  int64  
 10  Agency            148654 non-null  object 
dtypes: float64(6), int64(2), object(3)
memory usage: 12.5+ MB


- Se puede verificar que al borrar las columnas que no tenian solo *NaN* se disminuye el tamaño del dataframe a: *memory usage: 12.5+ MB*

**Observe los primeros 5 registros del DataFrame.**

In [ ]:
# Copie el código aca

*Los primeros 5 registros del dataframe son los siguientes*

In [ ]:
df.head(5)

,Id,EmployeeName,JobTitle,BasePay,OvertimePay,OtherPay,Benefits,TotalPay,TotalPayBenefits,Year,Agency
0,1,NATHANIEL FORD,GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY,167411.18,0.00,400184.25,NaN,567595.43,567595.43,2011,San Francisco
1,2,GARY JIMENEZ,CAPTAIN III (POLICE DEPARTMENT),155966.02,245131.88,137811.38,NaN,538909.28,538909.28,2011,San Francisco
2,3,ALBERT PARDINI,CAPTAIN III (POLICE DEPARTMENT),212739.13,106088.18,16452.60,NaN,335279.91,335279.91,2011,San Francisco
3,4,CHRISTOPHER CHONG,WIRE ROPE CABLE MAINTENANCE MECHANIC,77916.00,56120.71,198306.90,NaN,332343.61,332343.61,2011,San Francisco
4,5,PATRICK GARDNER,"DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)",134401.60,9737.00,182234.59,NaN,326373.19,326373.19,2011,San Francisco


**Observe los ultimos 5 registros del DataFrame.**

In [ ]:
# Copie el código aca

*Los últimos 5 registros del dataframe son los siguientes*

In [ ]:
df.tail(5)

,Id,EmployeeName,JobTitle,BasePay,OvertimePay,OtherPay,Benefits,TotalPay,TotalPayBenefits,Year,Agency
148649,148650,Roy I Tillery,Custodian,0.0,0.0,0.00,0.0,0.00,0.00,2014,San Francisco
148650,148651,Not provided,Not provided,NaN,NaN,NaN,NaN,0.00,0.00,2014,San Francisco
148651,148652,Not provided,Not provided,NaN,NaN,NaN,NaN,0.00,0.00,2014,San Francisco
148652,148653,Not provided,Not provided,NaN,NaN,NaN,NaN,0.00,0.00,2014,San Francisco
148653,148654,Joe Lopez,"Counselor, Log Cabin Ranch",0.0,0.0,-618.13,0.0,-618.13,-618.13,2014,San Francisco


**¿Cuantas filas y columnas tiene el dataframe?**

In [ ]:
# Copie el código aca

In [ ]:
df.shape

(148654, 11)

*El dataframe tiene 14864 filas y 11 columnas*

**Observe los 10 registros que estan desde el indice 3000**

In [ ]:
# Copie el código aca

*Los 10 registros del dataframe apartir del indice 3000 son los siguientes:*

In [ ]:
df.iloc[3000:].head(10)

,Id,EmployeeName,JobTitle,BasePay,OvertimePay,OtherPay,Benefits,TotalPay,TotalPayBenefits,Year,Agency
3000,3001,WILLIAM BOYLE,FIREFIGHTER,105934.65,18530.98,16389.60,NaN,140855.23,140855.23,2011,San Francisco
3001,3002,ALAN LIN,ELECTRONIC MAINTENANCE TECHNICIAN,98965.80,41504.40,380.00,NaN,140850.20,140850.20,2011,San Francisco
3002,3003,WILLIAM CONLEY,SERGEANT III (POLICE DEPARTMENT),130457.78,3131.42,7256.39,NaN,140845.59,140845.59,2011,San Francisco
3003,3004,ROBERT MCDONALD,"LIEUTENANT, FIRE DEPARTMENT",123104.99,1833.56,15897.68,NaN,140836.23,140836.23,2011,San Francisco
3004,3005,MICHAEL COSTELLO,POLICE OFFICER III,112401.08,7957.74,20466.93,NaN,140825.75,140825.75,2011,San Francisco
3005,3006,EDNA HO,MANAGER V,140821.37,0.00,0.00,NaN,140821.37,140821.37,2011,San Francisco
3006,3007,RODRIGO CONEL,REGISTERED NURSE,126120.00,3981.78,10714.21,NaN,140815.99,140815.99,2011,San Francisco
3007,3008,VICTOR HURTADO,FIREFIGHTER,105934.67,13112.76,21763.50,NaN,140810.93,140810.93,2011,San Francisco
3008,3009,DONALD LANDRY,TRANSIT POWER LINE WORKER,92861.50,39789.32,8135.86,NaN,140786.68,140786.68,2011,San Francisco
3009,3010,FRANK LEE,"INSPECTOR III, (POLICE DEPARTMENT)",130481.56,725.40,9579.58,NaN,140786.54,140786.54,2011,San Francisco



La columna *Benefits* tiene valores `NaN`, ¿Cuantos NaN tiene la columna *Benefits*? (TIP: Seleccione la Columna y vefique cuantos Na tiene `.isna()` luego sumelos `.sum()` )

In [ ]:
# Copie el código aca

In [ ]:
NanBen = df["Benefits"].isna().sum()
print(NanBen)

36163


Al parecer los valores faltantes en la columna Benefits es por que no se le pagan beneficios entonces:

**Reemplace los valores *NaN* de la Columna *Benefits* por el valor de cero (Recuerde usar el método de forma inplace) y mostrar los primeros 10 elementos de la lista**

In [ ]:
# Copie el código aca

In [ ]:
df.fillna({'Benefits': 0}, inplace=True)
df.head(10)

,Id,EmployeeName,JobTitle,BasePay,OvertimePay,OtherPay,Benefits,TotalPay,TotalPayBenefits,Year,Agency
0,1,NATHANIEL FORD,GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY,167411.18,0.00,400184.25,0.0,567595.43,567595.43,2011,San Francisco
1,2,GARY JIMENEZ,CAPTAIN III (POLICE DEPARTMENT),155966.02,245131.88,137811.38,0.0,538909.28,538909.28,2011,San Francisco
2,3,ALBERT PARDINI,CAPTAIN III (POLICE DEPARTMENT),212739.13,106088.18,16452.60,0.0,335279.91,335279.91,2011,San Francisco
3,4,CHRISTOPHER CHONG,WIRE ROPE CABLE MAINTENANCE MECHANIC,77916.00,56120.71,198306.90,0.0,332343.61,332343.61,2011,San Francisco
4,5,PATRICK GARDNER,"DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)",134401.60,9737.00,182234.59,0.0,326373.19,326373.19,2011,San Francisco
5,6,DAVID SULLIVAN,ASSISTANT DEPUTY CHIEF II,118602.00,8601.00,189082.74,0.0,316285.74,316285.74,2011,San Francisco
6,7,ALSON LEE,"BATTALION CHIEF, (FIRE DEPARTMENT)",92492.01,89062.90,134426.14,0.0,315981.05,315981.05,2011,San Francisco
7,8,DAVID KUSHNER,DEPUTY DIRECTOR OF INVESTMENTS,256576.96,0.00,51322.50,0.0,307899.46,307899.46,2011,San Francisco
8,9,MICHAEL MORRIS,"BATTALION CHIEF, (FIRE DEPARTMENT)",176932.64,86362.68,40132.23,0.0,303427.55,303427.55,2011,San Francisco
9,10,JOANNE HAYES-WHITE,"CHIEF OF DEPARTMENT, (FIRE DEPARTMENT)",285262.00,0.00,17115.73,0.0,302377.73,302377.73,2011,San Francisco


**¿Cual es la nueva información del dataframe?**

In [ ]:
# Copie el código aca

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148654 entries, 0 to 148653
Data columns (total 11 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Id                148654 non-null  int64  
 1   EmployeeName      148654 non-null  object 
 2   JobTitle          148654 non-null  object 
 3   BasePay           148045 non-null  float64
 4   OvertimePay       148650 non-null  float64
 5   OtherPay          148650 non-null  float64
 6   Benefits          148654 non-null  float64
 7   TotalPay          148654 non-null  float64
 8   TotalPayBenefits  148654 non-null  float64
 9   Year              148654 non-null  int64  
 10  Agency            148654 non-null  object 
dtypes: float64(6), int64(2), object(3)
memory usage: 12.5+ MB


**Cuantos valores unicos hay en las columnas categoricas?**

In [ ]:
# Copie el código aca

In [ ]:
col_cat = df.select_dtypes(include=['object']).columns
df[col_cat].nunique()

,0
EmployeeName,110811
JobTitle,2159
Agency,1


- la columna **Agency** solamente tiene un valor unico, entonces no aporta informacion, por lo tanto se puede eliminar

**Elimine la columna *Agency* del dataframe**

In [ ]:
# Copie el código aca

In [ ]:
# Eliminación de columna Agency
df.drop(columns= ["Agency"], inplace=True)

**Cual es el promedio de BasePay ?**

In [ ]:
# Copie el código aca

In [ ]:
prom_pago = df['BasePay'].mean()

print(f"El promedio de la columna BasePay es: {prom_pago}")

El promedio de la columna BasePay es: 66325.4488404877


*El promedio de la BasePay es 66325,45*

**Cual es la cantidad mas alta de OvertimePay en el dataset ?**

In [ ]:
# Copie el código aca

In [ ]:
max_overtp = df['OvertimePay'].max()
print(f"El valor máximo de la columna OvertimePay es: {max_overtp}")

El valor máximo de la columna OvertimePay es: 245131.88


*El valor máximo de la columna OvertimePay es: 245.131,88*

**Realice una descripcion estadistica de las columnas numericas?**

In [ ]:
# Copie el código aca

In [ ]:
desc = df[['Id', 'Year']].describe()
print(desc)

                  Id           Year
count  148654.000000  148654.000000
mean    74327.500000    2012.522643
std     42912.857795       1.117538
min         1.000000    2011.000000
25%     37164.250000    2012.000000
50%     74327.500000    2013.000000
75%    111490.750000    2014.000000
max    148654.000000    2014.000000


*Las columnas numericas son Id y Year, número de identificación y año, ambas variables tienen la misma cantidad de datos 148.654, el año mínimo es 2011 y el máximo 2014.*

**En este dataset las variables `Id` y `Year` representan variables categorica, convertir estas columnas a su formato correcto**

In [ ]:
# Copie el código aca

In [ ]:
# Convertir las columnas Id y Year a tipo objeto (string)
df['Id'] = df['Id'].astype('category')
df['Year'] = df['Year'].astype('category')

# Verificar los nuevos tipos de datos
print(df.dtypes[['Id', 'Year']])

Id      category
Year    category
dtype: object


*Se convirtieron a categoricas, debido a que las variables numericas id y year, no serán para realizar operaciones aritméticas, por lo tanto, se requieren categóricas.*

**la variable `Year` es categorica ordinal, primero detecte cuales son los valores unicos que tiene la columna**

In [ ]:
# Copie el código aca

In [ ]:
# definir la columna Year en categórica ordinal
cols_categoricas_ord = ["Year"]

# ver los datos únicos de las columnas_categoricas_nom
for columna in cols_categoricas_ord:
    print(f"Valores únicos en {columna}: {df[columna].unique()}")

Valores únicos en Year: [2011, 2012, 2013, 2014]
Categories (4, int64): [2011, 2012, 2013, 2014]


**A partir de la información anterior convierta la columna `Year` en variable categorica ordinal y visualice la columna**

In [ ]:
# Copie el código aca

In [ ]:
# Definimos el orden lógico de los años (de menor a mayor)
year_ordenados = sorted(df['Year'].unique())

# Convertimos la columna a tipo categórico ordenado
df['Year'] = pd.Categorical(df['Year'], categories=year_ordenados, ordered=True)

# Muesta las categorías ordenadas
df['Year'].unique()

[2011, 2012, 2013, 2014]
Categories (4, int64): [2011 < 2012 < 2013 < 2014]

*La variable year tiene las categorías 2011, 2012, 2013 y 2014.*

**Realice nuevamente una descripcion estadistica de las columnas numericas?**

In [ ]:
# Copie el código aca

In [ ]:
df.select_dtypes(include=['number']).describe()

,BasePay,OvertimePay,OtherPay,Benefits,TotalPay,TotalPayBenefits
count,148045.000000,148650.000000,148650.000000,148654.000000,148654.000000,148654.000000
mean,66325.448840,5066.059886,3648.767297,18924.232839,74768.321972,93692.554811
std,42764.635495,11454.380559,8056.601866,17165.279495,50517.005274,62793.533483
min,-166.010000,-0.010000,-7058.590000,-33.890000,-618.130000,-618.130000
25%,33588.200000,0.000000,0.000000,0.000000,36168.995000,44065.650000
50%,65007.450000,0.000000,811.270000,23214.020000,71426.610000,92404.090000
75%,94691.050000,4658.175000,4236.065000,33468.980000,105839.135000,132876.450000
max,319275.010000,245131.880000,400184.250000,96570.660000,567595.430000,567595.430000


**¿Cuál es el título de trabajo de JOSEPH DRISCOLL? Nota: Utilice mayúsculas, de lo contrario, puede obtener una respuesta que no coincida.**

In [ ]:
# Copie el código aca

In [ ]:
# Filtro por el nombre JOSEPH DRISCOLL para traer la columna JobTitle
emp_joseph = df[df['EmployeeName'] == 'JOSEPH DRISCOLL']['JobTitle']

print(emp_joseph)

24    CAPTAIN, FIRE SUPPRESSION
Name: JobTitle, dtype: object


*El empleo de Joseph Driscoll es captain.*

**¿Cuanto gana JOSEPH DRISCOLL (incluyendo beneficios)?**

In [ ]:
# Copie el código aca

In [ ]:
# Filtro por el nombre JOSEPH DRISCOLL para traer la columna TotalPayBenefits
TPB_joseph = df[df['EmployeeName'] == 'JOSEPH DRISCOLL']['TotalPayBenefits']

print(TPB_joseph)

24    270324.91
Name: TotalPayBenefits, dtype: float64


*El pago total incluidos beneficios de Joseph Driscoll es de 270.324,91.*

**¿Cual es el nombre de la persona con mejor pago (incluyendo beneficios)?**

In [ ]:
# Copie el código aca

In [ ]:
# Filtro el valor máximo de lla columna TotalPayBenefits
TPB_maxima = df[df['TotalPayBenefits'] == df['TotalPayBenefits'].max()]

# Imprimo el nombre del empleado y el valor total incluido beneficios
print(TPB_maxima[['EmployeeName', 'TotalPayBenefits']])

     EmployeeName  TotalPayBenefits
0  NATHANIEL FORD         567595.43


*La persona con mayor pago total incluidos beneficios es Nathaniel Ford, con valor pago total incluido beneficios de 567.595,43*

**¿Cual es el nombre de la persona peor paga (incluyendo beneficios)? ¿Notan algo extraño sobre cuánto le pagan?**

In [ ]:
# Copie el código aca

In [ ]:
# Filtro el valor máximo de lla columna TotalPayBenefits
TPB_min = df[df['TotalPayBenefits'] == df['TotalPayBenefits'].min()]

# Imprimo el nombre del empleado y el valor total incluido beneficios
print(TPB_min[['EmployeeName', 'TotalPayBenefits']])

       EmployeeName  TotalPayBenefits
148653    Joe Lopez           -618.13


*La persona con menor pago total incluidos beneficios es Joe Lopez, con valor pago total incluido beneficios de -618,13. Es extraño el valor, pues esta en negativo, es como si hubiese quedado en deuda con la compañía después de descuentos, o existe un error en el valor.*

**Cuantos registros tienen un valor de TotalPayBeneficts menor que cero**

In [ ]:
# Copie el código aca

In [ ]:
# Cuenta cuántas filas cumplen la condición
cant_TPB_negativos = (df['TotalPayBenefits'] < 0).sum()

# Imprime la cantidad de valores negativos columna TPB
print(f"Hay {cant_TPB_negativos} registros con pago menor a cero.")

Hay 4 registros con pago menor a cero.


*Hay 4 registros de personas con ingresos totales incluidos beneficios menores a cero.*

**¿Cual es el promedio (mean) BasePay de todos los empleados por año? los años van entre (2011-2014)?** (Tip: Usar groupby)

In [ ]:
# Copie el código aca

In [ ]:
# Agrupamos por año y calculamos el promedio de la columna BasePay
prom_year = df.groupby('Year')['BasePay'].mean()

print(prom_year)

Year
2011    63595.956517
2012    65436.406857
2013    69630.030216
2014    66564.421924
Name: BasePay, dtype: float64


/tmp/ipykernel_3459/4263204788.py:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  prom_year = df.groupby('Year')['BasePay'].mean()


El promedio de Pago Base por año es:
**2011** = 63.595,956517:
**2012** = 65436.406857 :
**2013** = 69630.030216 :
**2014** = 66564.421924 *texto en cursiva*

**¿Cuantos titulos de trabajos (JobTitle) unicos existen?**

In [ ]:
# Copie el código aca

In [ ]:
# Devuelve un array con los nombres de los cargos
tit_uniq = df['JobTitle'].unique()
print(tit_uniq)

['GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY'
 'CAPTAIN III (POLICE DEPARTMENT)' 'WIRE ROPE CABLE MAINTENANCE MECHANIC'
 ... 'Conversion' 'Cashier 3' 'Not provided']


*Los títulos de empleos (cargos) unicos son: GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY, CAPTAIN III (POLICE DEPARTMENT) y WIRE ROPE CABLE MAINTENANCE MECHANIC.*

**¿Cuales son los 5 trabajos mas comunes?**

In [ ]:
# Copie el código aca

In [ ]:
# Cuenta las repeticiones de cada JobTitle y toma los primeros 5
top_5_emp = df['JobTitle'].value_counts().head(5)

print(top_5_emp)

JobTitle
Transit Operator                7036
Special Nurse                   4389
Registered Nurse                3736
Public Svc Aide-Public Works    2518
Police Officer 3                2421
Name: count, dtype: int64


*Los títulos de empleos (cargos) más comunes son: Transit Operator, Special NUrse, Registered Nurse, Public Svc Aide-Public Works y Police officer 3.*

**¿Cuántos puestos de trabajo Unicos existieron en 2013? (Ejemplo, los títulos de trabajo con solo una ocurrencia en 2013?)**

In [ ]:
# Copie el código aca

In [ ]:
# Filtro el dataframe año 2013
df_2013 = df[df['Year'] == 2013]

# 2. Contamos las ocurrencias del título del empleo
conteo_JT = df_2013['JobTitle'].value_counts()

# 3. Filtramos los nombres de los cargos que aparecen solo 1 vez
JT_uniq = conteo_JT[conteo_JT == 1].index

# 4. Visualizo la lista de unicos empleos 2013
lista_final = JT_uniq.unique()

print(f"Lista de cargos con una sola ocurrencia: {lista_final}")
print(f"Total: {len(lista_final)}")

Lista de cargos con una sola ocurrencia: Index(['Mayor', 'Adm, SFGH Medical Center', 'Controller',
       'Parking Hearing Examiner', 'Court Asst., Superior Court',
       'Administrative Analyst II', 'IS Operator-Journey',
       'Drug Court Coordinator', 'Public Safety Comm Tech',
       'Orthopedic Technician 1',
       ...
       'Arborist Technician Supv II', 'Animal Control Supervisor',
       'Chief of Police', 'Head Photographer', 'Dep Dir for Investments, Ret',
       'IS Technician Assistant', 'Chief, Fire Department', 'Braillist',
       'Victim & Witness Technician', 'Senior Assessment Clerk'],
      dtype='object', name='JobTitle', length=202)
Total: 202


**¿Cuántas personas tienen la palabra Chief en el título de su trabajo?** Apoyese en los métodos para manipular strings y en el método `.apply()`

In [ ]:
# Copie el código aca

In [ ]:
total_chiefs = df['JobTitle'].apply(lambda x: 'chief' in str(x).lower()).sum()
print(f"El total de personas con 'Chief' en el título de empleo: {total_chiefs}")


El total de personas con 'Chief' en el título de empleo: 627


*El total de personas con 'Chief' en el título de empleo es de 627*

Este código es valido, se recomienda hacer el código mas simple y agregar typing y documentación.

**Bono: ¿Existe una correlación entre la longitud del nombre del Título del trabajo y el Salario?**

crea una nueva columna donde se guarde la longitud del titulo del trabajo. tip: usar el método `.apply()`

In [ ]:
# Copie el código aca

In [ ]:
# Creo la nueva columna con la longitud del título del empleo, uso str(x) para por si hay valores Nulos
df['JobTitleLength'] = df['JobTitle'].apply(lambda x: len(str(x)))

# Visualizamos las primeras filas para verificar el resultado
print(df[['JobTitle', 'JobTitleLength']].head())

                                         JobTitle  JobTitleLength
0  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY              46
1                 CAPTAIN III (POLICE DEPARTMENT)              31
2                 CAPTAIN III (POLICE DEPARTMENT)              31
3            WIRE ROPE CABLE MAINTENANCE MECHANIC              36
4    DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)              44


Utilice el método `.corr()` de Pandas para calcular la correlacion

In [ ]:
# Copie el código aca

In [ ]:
# Calculo la correlación entre las 2 columnas: JobTitleLength y TotalPayBenefits
corr_JTL_TPB= df[['JobTitleLength', 'TotalPayBenefits']].corr()

print(corr_JTL_TPB)

                  JobTitleLength  TotalPayBenefits
JobTitleLength          1.000000         -0.036878
TotalPayBenefits       -0.036878          1.000000


*En la matriz de correlación observamos valores menores a cero, lo que nos indica que no existe correlación entre las variables (columnas) JobTitleLength y TotalPayBenefits, es decir, la longitud del título no incide en el salario o pago total incluido beneficios.*